internal/gatewayapi/translator.go

```diff
@@ -901,108 +1004,35 @@ func (t *Translator) ProcessHTTPRoutes(httpRoutes []*v1beta1.HTTPRoute, gateways
 				}
 
 				for _, backendRef := range rule.BackendRefs {
-					if backendRef.Group != nil && *backendRef.Group != "" {
-						parentRef.SetCondition(
-							v1beta1.RouteConditionResolvedRefs,
-							metav1.ConditionFalse,
-							v1beta1.RouteReasonInvalidKind,
-							"Group is invalid, only the core API group (specified by omitting the group field or setting it to an empty string) is supported",
-						)
-						continue
-					}
 
-					if backendRef.Kind != nil && *backendRef.Kind != KindService {
-						parentRef.SetCondition(
-							v1beta1.RouteConditionResolvedRefs,
-							metav1.ConditionFalse,
-							v1beta1.RouteReasonInvalidKind,
-							"Kind is invalid, only Service is supported",
-						)
-						continue
-					}
+					destination := buildRuleRouteDest(backendRef, parentRef, httpRoute, resources)
 
-					if backendRef.Namespace != nil && string(*backendRef.Namespace) != "" && string(*backendRef.Namespace) != httpRoute.Namespace {
-						if !isValidCrossNamespaceRef(
-							crossNamespaceFrom{
-								group:     v1beta1.GroupName,
-								kind:      KindHTTPRoute,
-								namespace: httpRoute.Namespace,
-							},
-							crossNamespaceTo{
-								group:     "",
-								kind:      KindService,
-								namespace: string(*backendRef.Namespace),
-								name:      string(backendRef.Name),
-							},
-							resources.ReferenceGrants,
-						) {
-							parentRef.SetCondition(
-								v1beta1.RouteConditionResolvedRefs,
-								metav1.ConditionFalse,
-								v1beta1.RouteReasonRefNotPermitted,
-								fmt.Sprintf("Backend ref to service %s/%s not permitted by any ReferenceGrant", *backendRef.Namespace, backendRef.Name),
-							)
-							continue
+					for _, route := range ruleRoutes {
+						// If the route already has a direct response or redirect configured, then it was from a filter so skip
+						// processing any destinations for this route.
+						if route.DirectResponse == nil && route.Redirect == nil {
+							if destination != nil {
+								route.Destinations = append(route.Destinations, destination)
+								continue
+							} else {
+								route.InvalidBackends += 1
```

Comment:
should `InvalidBackends` be a field within `DirectResponse` ?

Reply:
No because it's used to calculate the weights. If we're using `DirectResponse` then we don't do anything with the weight, but I'm going to need to rework that field anyways to properly calculate the weight.

Reply:
If `DirectResponse` did support weighted matches alongside regular clusters that would be great since I wouldn't have to setup the weighted clusters.

Reply:
if the below approach of using the basic `cluster` works, can we reuse `RouteDestination` https://github.com/envoyproxy/gateway/blob/54b26e2ef5d2bb6d536539acf0cbbfa5cdc31bea/internal/ir/xds.go#L219 and explicitly set an `Invalid` bool in there ?